static install failed after built success

CMake Error at Applications/src/cmake_install.cmake:54 (file):
  file RPATH_CHANGE could not write new RPATH:

    $ORIGIN/..

  to the file:

    /rds/projects/d/duanj-ai-in-medical-imaging/castles/.share/2020a-gcc9.3/library/mritk-2.0.0/lib/mirtk/tools/info

  No valid ELF RPATH or RUNPATH entry exists in the file;
Call Stack (most recent call first):
  Applications/cmake_install.cmake:178 (include)
  cmake_install.cmake:203 (include)


make: *** [Makefile:74: install] Error 1

--- a/CMake/Modules/mirtkAddExecutable.cmake
+++ b/CMake/Modules/mirtkAddExecutable.cmake
@@ -93,16 +93,18 @@ function(mirtk_add_executable target_name)
     # used by basis_set_target_install_rpath to work. This function is very
     # costly. Hence, as we know what RPATH we want without inspecting the list
     # of all link dependencies, we rather set the INSTALL_RPATH ourselves.
-    if (CMAKE_HOST_APPLE)
-      set(ORIGIN "@loader_path")
-    else ()
-      set(ORIGIN "\$ORIGIN")
+    if (BUILD_SHARED_LIBS)
+      if (CMAKE_HOST_APPLE)
+        set(ORIGIN "@loader_path")
+      else ()
+        set(ORIGIN "\$ORIGIN")
+      endif ()
+      basis_get_relative_path(rpath
+        "${CMAKE_INSTALL_PREFIX}/${INSTALL_LIBEXEC_DIR}"
+        "${CMAKE_INSTALL_PREFIX}/${INSTALL_LIBRARY_DIR}"
+      )
+      string(REGEX REPLACE "/+$" "" rpath "${rpath}")
+      set_target_properties(${target_uid} PROPERTIES INSTALL_RPATH "${ORIGIN}/${rpath}")
     endif ()
-    basis_get_relative_path(rpath
-      "${CMAKE_INSTALL_PREFIX}/${INSTALL_LIBEXEC_DIR}"
-      "${CMAKE_INSTALL_PREFIX}/${INSTALL_LIBRARY_DIR}"
-    )
-    string(REGEX REPLACE "/+$" "" rpath "${rpath}")
-    set_target_properties(${target_uid} PROPERTIES INSTALL_RPATH "${ORIGIN}/${rpath}")
   endif ()
 endfunction()

